### ISSUE TITLE
Backdrop database update "Updates were attempted" is not very reassuring.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#638]
We currently say:

> **Backdrop database update**
> 
> Updates were attempted. If you see no failures below, you may proceed happily back to your site. Otherwise, you may need to update your database manually. All errors have been [logged](/admin/reports/dblog).

Isn't there any way to detect if there were no failures and be more confident about it. Like so:

> **Backdrop database update**
> 
> Updates completed successfully. You may proceed happily back to your site. Any errors have been [logged](/admin/reports/dblog).

**Comment by klonos** [comment]
...and if there were actually failures:

> **Backdrop database update**

> Updates were attempted but there were some failures:

> ...[list of failures]

> You may need to update your database manually. All errors have been [logged](/admin/reports/dblog).

**Comment by quicksketch** [comment]
LOL, this is a great suggestion. I've always been slightly worried by the "Updates where attempted" line as well. I think it's because our code simply doesn't check if there were errors or not before it displays that message, so it tries to account for both successes and failures with the same sentence. Let's do the finger-work to add that check.

**Comment by serundeputy** [comment]
In `update.php` i see checks for success:
```php
if ($_SESSION['update_success']) {
    $output = '<p>Updates were attempted. If you see no failures below, you may proceed happily back to your <a href="' . base_path() . '">site</a>. Otherwise, you may need to update your database manually.' . $log_message . '</p>';
  }
  else {
    $updates_remaining = reset($_SESSION['updates_remaining']);
```

the only other instances of "Updates were attempted" are in *.test files.

Do we just need to change the markup or is there more to it than that?

~Geoff

**Comment by klonos** [comment]
I guess it would be safe to throw a "success!" message if ```$log_message``` is empty.

I realize that some developers put information text on success there, so we can't really be sure if ```$log_message``` is not empty that the message was an error/failure or not.

**Comment by quicksketch** [comment]
Huh, sure enough. Let's try making an update that throws an error and see what the result is. You could do something simple like `db_drop_table('gibberish')` in an update hook, and that will cause it to fail when the table isn't found. I think you still get to the success page anyway, but the error is displayed.

**Comment by klonos** [comment]
> ...you still get to the success page anyway, but the error is displayed.

Yep, that's the point. Ending up in the same finish page, but the text will be different depending on whether there were errors or not. The tricky part here would be to distinguish between errors/failures and simple "everything ok" log messages stored in ```$log_message```.

**Comment by klonos** [comment]
It would be so much easier if there were 3 variations like ```$log_message```, ```$log_warning``` and ```$log_error```.

**Comment by klonos** [comment]
...or have a way to define the message level to be either *status*, *warning* or *error*. Pretty much like we do for the site status report page I guess.
